adapters/algorix/algorixtest/exemplary/sample-banner-with-mediatype.json

```diff
@@ -0,0 +1,105 @@
+{
+  "mockBidRequest": {
+    "id": "test-request-id",
+    "imp": [
+      {
+        "id": "test-imp-id",
+        "banner": {
+          "format": [{"w": 320, "h": 50}]
+        },
+        "video": {
+          "mimes": [
+            "video/mp4"
+          ],
+          "w": 320,
+          "h": 480
+        },
+        "ext": {
+          "bidder": {
+            "sid": "testSid",
+            "token": "testToken",
+            "region": "APAC"
+          }
+        }
+      }
+    ]
+  },
+
+  "httpCalls": [
+    {
+      "expectedRequest": {
+        "uri": "https://apac.xyz.test.com?sid=testSid&token=testToken",
+        "body": {
+          "id": "test-request-id",
+          "imp": [
+            {
+              "id":"test-imp-id",
+              "banner": {
+                "format": [{"w": 320, "h": 50}],
+                "w": 320,
+                "h": 50
+              },
+              "video": {
+                "mimes": [
+                  "video/mp4"
+                ],
+                "w": 320,
+                "h": 480
+              },
+              "ext": {
+                "bidder": {
+                  "sid": "testSid",
+                  "token": "testToken",
+                  "region": "APAC"
+                }
+              }
+            }
+          ]
+        }
+      },
+      "mockResponse": {
+        "status": 200,
+        "body": {
+          "id": "test-request-id",
+          "seatbid": [
+            {
+              "seat": "ttx",
+              "bid": [{
+                "id": "8ee514f1-b2b8-4abb-89fd-084437d1e800",
+                "impid": "test-imp-id",
+                "price": 0.5,
+                "adm": "some-ads",
+                "crid": "crid_testid",
+                "ext": {
+                  "mediaType": "video"
```

Comment:
How to pass the code test? if return a "banner" media type, the code must return the "banner", not "video", otherwise the test failed. Or add Exception to test return mediaType and request format error, but why we need the response mediaType, only direct use bidRequest format.

Why we add mediaType, when I first commit AlgoriX Bidder Adapter, it comment: Does your bidding server include the bid type in it's response? it seems mediaType suggest from bidResponse not bid Request.

Reply:
Your old code, when it sees a multiformat impression, will assume you are always bidding a video ad. If, at some point in time, your bidder may decide to bid a banner instead of video, we want to ensure the bid is marked properly. So here we want the mock response to report that it is providing a banner ad. The way this test is currently crafted, the test will pass both before and after the PR is applied, so it does not directly test the code in the PR.

For example, if someone in a future PR introduces a typo into the definition of `algorixResponseBidExt`, such that it does not properly parse the mediaType, then your new code will be skipped (mediaType will not match any of the options in the case statement) and you will drop down into the old code. The test here will still pass however, since that is the media type it will assume if the parameter doesn't exist in your bid. Better to have a test that will catch the case where the new code breaks, so we can ensure it does not happen in the final release.

Reply:
Ok, I know.
I add an exception for MediaType test. if multi-format request and not return mediaType or mediaType is bad. it will return an Exception for unable fetch MediaType.

Reply:
No I don't think you understand. By "test" I mean a JSON test that verifies that the new code will read the media type from the response, rather than returning a media type based on the old logic. We did not need a test in your adapter code itself. It doesn't hurt to have this extra test, but things would need to be very broken for both the new code and the old code to fail to come up with a media type.

I was just suggesting, in this file, to change the mediaType here to be banner, and below in the expected bid response to be banner as well. But I suppose your test below for "sample-banner-with-other-mediatype.json" will also test this case.

Reply:
means add a video bidRequest test, and Response return banner to test it?

Reply:
Yes, now just move the other file to supplemental and I can approve this.